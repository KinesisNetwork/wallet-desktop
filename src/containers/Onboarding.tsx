import { closeModal, completeOnBoarding } from '@actions'
import { RootState } from '@store';
import * as React from 'react'
import { connect } from 'react-redux'

import * as onboarding from '@images/onboarding_stickers_Artboard.svg'

const mapStateToProps = (state: RootState) => ({
  isModalActive: state.modals.modalDisplay,
  hasOnBoarded: state.settings.onBoarding
})

const mapDispatchToProps = {
  closeModal,
  completeOnBoarding
}

type Props = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps

export class OnboardingPresentation extends React.Component<Props> {
  closeModal = () => {
    this.props.closeModal()
    this.props.completeOnBoarding()
  }

  render() {
    return (
      <main className={`modal ${this.props.isModalActive && !this.props.hasOnBoarded ? 'is-active' : ''}`}>
        <div className="modal-background" />
        <section className="modal-content has-background-grey-dark section" style={{ borderRadius: '5px', width: '60rem' }}>
          <h1
            className="is-size-3 has-text-weight-semibold has-text-grey-lighter has-text-centered u-margin-bottom-lg-2"
            style={{ letterSpacing: '1px' }}
          >
            Your new wallet is now active
          </h1>
          <div className="columns is-vcentered u-margin-bottom-lg-2">
            <div className="column">
              <div className="content has-text-grey-lighter has-text-weight-semibold is-size-5">
                <p style={{ letterSpacing: '3px' }}>Buy Kinesis from the exchange and store it in your wallet.</p>
                <p style={{ letterSpacing: '3px' }}>Kinesis is the world's most stable currency, backed 1:1 against gold (KAU) and silver (KAG) bullion.</p>
                <a className="is-link is-size-6">Learn more</a>
              </div>
            </div>
            <div className="column">
              <figure className="image is-4by3">
                <img src={onboarding ? onboarding : './onboarding_stickers_Artboard.svg'} className="" />
              </figure>
            </div>
          </div>
          <div className="is-pulled-right">
            <button
              className="button has-text-grey-darker is-uppercase is-primary"
              onClick={() => this.closeModal()}
            >OK</button>
          </div>
        </section>
      </main>
    )
  }
}

const ConnectedOnboardingPresentation = connect(
  mapStateToProps,
  mapDispatchToProps
)(OnboardingPresentation)

export { ConnectedOnboardingPresentation as Onboarding }
