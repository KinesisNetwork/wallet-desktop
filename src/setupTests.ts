import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
