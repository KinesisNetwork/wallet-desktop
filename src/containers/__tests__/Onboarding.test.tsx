import { closeModal, completeOnBoarding } from '@actions'
import { shallow } from 'enzyme'
import * as React from 'react'

import { OnboardingPresentation } from '@containers/Onboarding'
import '../../setupTests'

describe('OnboardingPresentation', () => {
  const props = {
    closeModal,
    completeOnBoarding,
    isModalActive: true,
    hasOnBoarded: false,
  }

  it('should render correctly', () => {
    const wrapper = shallow(<OnboardingPresentation {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should not render modal if the onboarding has already been done', () => {
    const wrapper = shallow(<OnboardingPresentation {...props} hasOnBoarded={true} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('calls closeModal and completeOnBoarding methods on clicking the button', () => {
    const closeModalStub = jest.fn()
    const completeOnBoardingStub = jest.fn()
    const wrapper = shallow(
      <OnboardingPresentation
        {...props}
        closeModal={closeModalStub}
        completeOnBoarding={completeOnBoardingStub}
      />,
    )

    const button = wrapper.find('button')
    button.simulate('click')

    expect(closeModalStub).toHaveBeenCalled()
    expect(completeOnBoardingStub).toHaveBeenCalled()
  })
})
