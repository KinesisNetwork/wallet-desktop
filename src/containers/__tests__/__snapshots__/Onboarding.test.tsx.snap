// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingPresentation should not render modal if the onboarding has already been done 1`] = `
<Modal
  hasBackground={true}
  isActive={false}
>
  <div
    className="box"
  >
    <h1
      className="title is-2 has-text-primary is-uppercase has-text-centered"
    >
      Your new wallet is now active
    </h1>
    <div
      className="columns is-vcentered"
    >
      <div
        className="column"
      >
        <div
          className="content has-text-grey-lighter is-size-4"
        >
          <p>
            Buy Kinesis from the exchange and store it in your wallet.
          </p>
          <p>
            Kinesis is the world's most stable currency, backed 1:1 against gold (KAU) and silver (KAG) bullion.
          </p>
          <a
            className="is-link is-size-6"
            href="https://youtu.be/moDBJCemGTM"
            target="_blank"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        className="column"
      >
        <figure
          className="image is-4by3"
        >
          <img
            className=""
            src="test-file-stub"
          />
        </figure>
      </div>
    </div>
    <div
      className="field is-grouped is-grouped-right"
    >
      <div
        className="control"
      >
        <button
          className="button is-primary"
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`OnboardingPresentation should render correctly 1`] = `
<Modal
  hasBackground={true}
  isActive={true}
>
  <div
    className="box"
  >
    <h1
      className="title is-2 has-text-primary is-uppercase has-text-centered"
    >
      Your new wallet is now active
    </h1>
    <div
      className="columns is-vcentered"
    >
      <div
        className="column"
      >
        <div
          className="content has-text-grey-lighter is-size-4"
        >
          <p>
            Buy Kinesis from the exchange and store it in your wallet.
          </p>
          <p>
            Kinesis is the world's most stable currency, backed 1:1 against gold (KAU) and silver (KAG) bullion.
          </p>
          <a
            className="is-link is-size-6"
            href="https://youtu.be/moDBJCemGTM"
            target="_blank"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        className="column"
      >
        <figure
          className="image is-4by3"
        >
          <img
            className=""
            src="test-file-stub"
          />
        </figure>
      </div>
    </div>
    <div
      className="field is-grouped is-grouped-right"
    >
      <div
        className="control"
      >
        <button
          className="button is-primary"
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  </div>
</Modal>
`;
